tigerblood.go

```diff
@@ -285,7 +287,13 @@ func (h *TigerbloodHandler) UpsertReputationByViolation(w http.ResponseWriter, r
 		return
 	}
 
-	err = h.db.InsertOrUpdateReputationEntryByViolationType(nil, ip, entry.Violation)
+	// lookup violation weight in config map
+	var penalty, ok = h.violationPenalties[entry.Violation]
+	if !ok {
+		log.Printf("Could not find violation type: %s", entry.Violation)
+		penalty = 0
```

Comment:
In the case where no violation is found, I would return a 400 instead of executing a query that will effectively not change the reputation. This way we save a round trip to the database and the client can know they're doing something wrong.